### PR TITLE
Eng 23547 data race 1

### PR DESCRIPTION
**Pull request by rdykiel**
Eliminate data races uncovered by ginkgo race detection.

### CODE REVIEW

File: voltdbclient/distributor.go [hunk 1]
```diff
@@ -105,6 +109,17 @@ func newConn(cis []string, duration time.Duration) (*Conn, error) {
 	return c, nil
 }
 
+// setupAffinity sets up the response channels before the Conn is started to avoid race in loop()
+func (c *Conn) setupAffinity() {
+	if !c.useClientAffinity {
+		return
+	}
+
+	c.subTopoCh = make(chan voltResponse, 1)
```
Comment (akhanzode) [inline voltdbclient/distributor.go:118]: These should be closed on connection close.

**Review by dave-voltdb — APPROVED** [review]
Looks ok to me.

File: voltdbclient/distributor.go (hunk 1 above)
Comment (rdykiel) [reply voltdbclient/distributor.go:118]: Please note the following:
- The life-cycle of those channels haven't changed: they are created if affinity is true (it is always true BTW), and handled by the loop() method. They are simply instantiated earlier in the life-cycle of the Conn object to avoid the race in loop()
- It would be dangerous to add close() statements to those channels on connection close, because you have no way of testing if a channel has been closed by somebody. There was no close() in the pre-existing code and I don't think they are necessary.
- The call to close() on a channel is optional and does not result in a leak (not like goroutines, which can leak).

Bottom line: I don't want to add code that didn't exist before, and that risks a panic in execution.

**Review by akhanzode — APPROVED** [review]
